src/Functions/parseDateTime.cpp

```diff
@@ -598,13 +610,71 @@ namespace
             validateFunctionArguments(*this, arguments, mandatory_args, optional_args);
 
             String time_zone_name = getTimeZone(arguments).getTimeZone();
-            DataTypePtr date_type = std::make_shared<DataTypeDateTime>(time_zone_name);
+            DataTypePtr date_type = nullptr;
+            if constexpr (parseDateTime64)
+            {
+                String format = getFormat(arguments);
+                std::vector<Instruction> instructions = parseFormat(format);
+                UInt32 scale = 0;
+                if (!instructions.empty())
+                {
+                    for (const auto & ins : instructions)
+                    {
+                        if (scale > 0)
+                            break;
+                        const String fragment = ins.getFragment();
+                        for (char ch : fragment)
+                        {
+                            if (ch != 'S')
+                            {
+                                scale = 0;
```

Comment:
The code makes the assumption that the format string contains just a single sequence of `S`s. We should at least document this.